File: Array/liuchang/day44/day44.java
public class day44 {

    public static int firstUniqChar(String str) {
        int length = str.length();
//        printf("%d\n", length);
        int n = 0;
        char chars[] = str.toCharArray();
        while (chars[n]!='\0' ) {
            int index = n;
            for (int i = 0; i < length; ++i) {
                if (chars[index] == chars[i] && i != index) {
                    break;
                } else if (i == (length - 1)) {
                    return n;
                }
            }
            index++;
            n++;
        }
        return -1;
    }

    public static void main(String[] args) {
        System.out.println("====="+firstUniqChar("loveleetcode"));
    }
}

File: Array/liuchang/day44/day44.kt
package net.study.kotlintest

/**
 * @author       刘畅
 * @createdate   2020/7/14
 * @describe
 */
class day44 {
    fun firstUniqChar(str: String): Int {
        var length = str.length;
//        printf("%d\n", length);
        var n = 0;
        var chars = str.toCharArray();
        while (chars[n] != null) {
            var index = n;
            for (i in 0 until length){
                if (chars[index] == chars[i] && i != index) {
                    break;
                } else if (i == (length - 1)) {
                    return n;
                }
            }
            index++;
            n++;
        }
        return -1;
    }

    fun main(args: Array<String>) {
        System.out.println("=====" + firstUniqChar("loveleetcode"));

    }
}